### ISSUE TITLE
Create new module command not found

### ISSUE THREAD

**Issue by CES-dengzeyuan** [tlaplus/vscode-tlaplus#269]
Hi - When I followed the Getting Started, I couldn't find either "Create new module (TLA+)" or "Create PlusCal block (TLA+)" snippet from the drop-down list. 

<img width="623" alt="截屏2022-09-15 14 32 45" src="https://user-images.githubusercontent.com/46039524/190331201-40059438-87a3-4079-964d-54982e06608c.png">

So I wonder how can I get/generate a program demo before I go ahead.

Thanks a lot.

**Comment by nano-o** [comment]
This indeed seems to be confusing for new users.

**Comment by lemmy** [comment]
Yes, this is annoying.  It looks like the trouble you're facing might be because VSCode loads the TLA+ extension only after opening a `.tla` file. So, when you have a fresh workspace without any `.tla` files, the TLA+ extension isn't activated, and that's why the VSCode command emmet doesn't display the TLA+ commands.

**Comment by nano-o** [comment]
Those commands also don't appear in the command menu even from a loaded tla file on my setup. I'm using the nightly version of the extension.

**Comment by lemmy** [comment]
Is the editor with the .tla file VSCode's active editor?

**Comment by nano-o** [comment]
I think so. That's the only open file and my cursor is in it. It says TLA+ at the bottom right of the window. But then I don't know much about VS Code.

**Comment by lemmy** [comment]
Can you post a screenshot of the VSCode window with the command emmet expanded?

**Comment by nano-o** [comment]
Is that what you mean by "command emmet expanded"?

**Comment by lemmy** [comment]
Thanks, yes, that's what I meant.  In the screenshot, the TLA+ commands do show up.  When are they missing?

**Comment by nano-o** [comment]
That's what missing: "Create new module (TLA+)" or "Create PlusCal block (TLA+)"

**Comment by lemmy** [comment]
Ohh, those two commands also don't show up for me.  I have the nightly build of the VSCode extension (not the released one) installed.

**Comment by FedericoPonzi** [comment]
Please check https://docs.tlapl.us/using:vscode:getting_started
The feature mentioned there is for snippets (ref: https://code.visualstudio.com/docs/editing/userdefinedsnippets). 
you need to either create a .tla file, or create a new empty file:

![Image](https://github.com/user-attachments/assets/353268d5-5f59-4d7c-83c8-7bdc96131bb7)
click on "select a language" and select TLA+.  Then start typing `module`:

![Image](https://github.com/user-attachments/assets/de49a5c0-cbb9-41c0-990f-ea1cfa5a2843)
From this menu, if you select `module` it will automatically generate the header and footer for you. Similar snippet is available for `pluscal`.
